app/src/main/java/com/android/partagix/ui/screens/HomeScreen.kt

```diff
@@ -1,40 +1,149 @@
-            modifier = modifier.testTag("homeScreenBottomNavBar"))
+            modifier = Modifier.testTag("homeScreenBottomNavBar"))
       }) { innerPadding ->
         Column(
-            modifier = Modifier.fillMaxSize().padding(innerPadding),
-            verticalArrangement = Arrangement.Center,
-            horizontalAlignment = Alignment.CenterHorizontally) {
+            modifier =
+                Modifier.fillMaxSize().padding(innerPadding).testTag("homeScreenMainContent")) {
+              HorizontalDivider(modifier = Modifier.fillMaxWidth())
               Text(
-                  text = "Home Screen coming soon!",
-                  style = TextStyle(fontFamily = FontFamily.Monospace, fontSize = 24.sp),
-                  modifier = Modifier.testTag("homeScreenMainContent"))
+                  text = "Quick access",
+                  modifier = Modifier.fillMaxWidth().padding(start = 16.dp, top = 16.dp),
+                  style = MaterialTheme.typography.titleLarge)
+              Spacer(modifier = Modifier.height(8.dp))
+              Row(
+                  modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 8.dp),
+                  horizontalArrangement = Arrangement.SpaceBetween) {
+                    BigButton(
+                        logo = Icons.Default.PersonSearch,
+                        text = "Find item to borrow",
+                        onClick = {},
+                        modifier = Modifier.weight(1f).testTag("homeScreenFirstBigButton"))
+                    Spacer(modifier = Modifier.width(8.dp))
+                    BigButton(
+                        logo = Icons.Default.QrCodeScanner,
+                        text = "Quick scan",
+                        onClick = {},
+                        modifier = Modifier.weight(1f).testTag("homeScreenSecondBigButton"))
+                    Spacer(modifier = Modifier.width(8.dp))
+                    BigButton(
+                        logo = Icons.Default.ImageSearch,
+                        text = "Find item in inventory",
+                        onClick = {},
+                        modifier = Modifier.weight(1f).testTag("homeScreenThirdBigButton"))
+                  }
+              Text(
+                  text = "New borrowing requests",
+                  modifier = Modifier.fillMaxWidth().padding(start = 16.dp, top = 16.dp),
+                  style = MaterialTheme.typography.titleLarge)
+              ItemListColumn(
+                  List =
+                      uiState.borrowedItems, // TODO replace this with the actual list of borrowing
+                  // requests
+                  users = uiState.usersBor,
+                  loan = uiState.loanBor,
+                  Title = "",
+                  corner = "", // TODO this should be a button to the full borrowing screen
+                  onClick = { /* TODO navigate to borrow request */},
+                  onClickCorner = { /* TODO */},
```

Comment:
You could expand on what is to be done here in case someone else needs to continue working on this code.

Reply:
addressed in following commits